mappymatch/utils/plot.py

```diff
@@ -178,3 +192,91 @@ def plot_map(tmap: NxMap, m=None):
         ).add_to(m)
 
     return m
+
+
+def plot_match_distances(coord_df):
+    # todo MatchResult is not a defined Element, removed from the args list above (was (matches: MatchResult)).
+    # build and display plots here.
+    """
+    Summary: Plot the points deviance from known roads with matplotlib.
+
+    Issues:
+        we have two different dataframes one labeled with gdf and the other with df. --> to resolve this, the gdf labeled data frames have been changed to df labels.
+
+    Args:
+        coord_df (pandas dataframe): coords of guessed points in the area.
+    """
+
+    # Road Data Frame Section
+    # Define a pandas data frame containing the list of matches (each represented by m) where m.road = True
+
+    # mid_i = int(
+    #     len(coord_df) / 2
+    # )  # find the middle index of the coord_gdf data frame.
+    # mid_coord = coord_df.iloc[
+    #     mid_i
+    # ].geometry  # answer the question: what is the middle coordinate?
+
+    y = coord_df.distance  # the distances from the expected line. Deviance.
+    x = [x for x in range(0, len(y))]  # create blanks for x axis
+
+    # for (
+    #     coord
+    # ) in (
+    #     coord_df.itertuples()
+    # ):  # for every coordinate tuple within coord_gdf ...
+    #     x_coord = coord.geometry.x  # identify the x coordinate geometry.
+    #     y_coord = coord.geometry.y  # identify the y coordinate geometry.
+
+    # for (
+    #     road
+    # ) in road_df.itertuples():  # for every road in the road_gdf data frame...
+    #     full_line = [
+    #         (lat, lon) for lon, lat in road.geometry.coords
+    #     ]  # identify the full line of that road in lat,long tuples.
+
+    # Plotting Section
+    plt.figure(figsize=(15, 7))  # create a figure sized 15 x 7
+    plt.autoscale(
+        enable=True
+    )  # autoscale the figure's contents to the data once it is plotted.
+    # todo ---- what was the purpose of plt.stem? -->  plt.stem(x, y)
+    plt.scatter(
+        x, y
+    )  # create a scatter plot of our x (blanks), and our y (deviance from expected line) values.
+    plt.title("Distance To Nearest Road")  # create a title for our plot.
+    plt.ylabel("Meters")  # establish the y axis label "Meters"
+    plt.xlabel(
+        "Point Along The Path"
+    )  # label the x axis label "Point Along The Path"
+    plt.show()  # print the plot.
+
+
+def plot_prep(file_path):  #
```

Comment:
I think we can remove this function as we assume the user will already have these objects built when using any of the plotting utilities.